components/rankings/rankingsController.js

```diff
@@ -1,142 +1,172 @@
-import errors from '../errors'
+import moment from 'moment'
+
 import service from './rankingsService'
 import utils from './rankingsUtils'
-import dal from './rankingsDAL'
+import interactions from '../interactions'
 import users from '../users'
 import messages from '../messages'
 
-const file = 'Ranking | Controller'
-
-const calculatePositionByUser = async (user, isCoreTeam) => {
-  return service.calculatePositionByUser(user, isCoreTeam)
-}
-
-const commandGeneral = async message => {
-  const coreTeam = await users.isCoreTeam(message.u._id)
-  return service.getGeneralRanking(message.u._id, coreTeam)
-}
-
-const commandByMonth = async message => {
-  const month = await utils.getMonthFromMessage(message)
-  const user = await users.findOne({ rocketId: message.u._id })
-  return service.getRankingMessageByMonth(user, month)
-}
-
-const getRankingByMonth = async (month, team = false) => {
-  if (!(await utils.isValidMonth(month))) month = utils.getCurrentMonth()
-
-  const ranking = await service.getRankingByMonth(month)
-  const monthName = utils.getMonthName(month - 1)
-
-  const data = {
-    first_users: [],
-    last_users: [],
-    monthName: monthName,
-    error: null
-  }
-
-  if (ranking.error) {
-    data.error = ranking.error
-    return
-  }
+const sendToChannel = async () => {
+  const roomname = process.env.ROCKET_DEFAULT_CHANNEL
+  const isEnabled = process.env.ROCKET_SEND_TO_CHANNEL
 
-  if (!ranking.users.length) {
-    data.error = `Ops! Ainda ninguém pontuou em ${monthName}. =/`
-    return
-  }
+  if (!roomname || !isEnabled) return
+  const ranking = await interactions.findByDate({})
 
-  if (ranking.users.length < 3) {
-    data.error = `Ops! Ranking incompleto em ${monthName}. =/`
-    return
+  if (ranking.error || ranking.length < 5) return
+  let response = {
+    msg: `Saiba quem são as pessoas que mais me orgulham no Olímpio pela interação.
+Essas nobres pessoas têm se destacado em meu templo:`,
+    attachments: []
   }
 
-  const rankingUsers = await service.generateUsersPosition(ranking.users, team)
-
-  data.first_users = rankingUsers.slice(0, 3)
-  data.last_users = rankingUsers.slice(3, 20)
+  response.attachments = await ranking.slice(0, 5).map((user, index) => {
+    return {
+      text: `${index + 1}º lugar está ${user.name} com ${
+        user.monthlyScore
+      } xp e nível ${user.level}`
+    }
+  })
 
-  return data
+  return await messages.sendToRoom(response, roomname)
```

Comment:
```suggestion
  return messages.sendToRoom(response, roomname)
```

O correto aqui é vc retornar o messages.sentToRoom direto, assim estará retornando a promise, e usa o await direto no método que está consumindo

Reply:
Nao vi necessidade,  mas alterei la. qualquer coisa a gente troca uma ideia depois

Reply:
Mais questão de redundância mesmo, pois mesmo no método que consome vc precisa usar o await ou um then, seria o mesmo que faze rum .then(...).then(...), funciona, mas não é uma boa prática

Reply:
Ok, vlw!